tests/test_occupationcoder.py

```diff
@@ -2,43 +2,162 @@
+            .apply(lambda x: cl.simple_clean(x, known_only=False))
+        df['clean_desc'] = df['job_description']\
+            .apply(lambda x: cl.simple_clean(x, known_only=False))
 
-    def test_001_running_on_a_file(self):
+        for index, row in df.iterrows():
+            clean = " ".join([row['clean_title'],
+                              row['clean_sector'],
+                              row['clean_desc']])
+            SOC_codes = self.matcher.get_tfidf_match(clean)
+            for code in SOC_codes:
+                self.assertIn(code, self.expected_codes)
+
+    def test_code_record(self):
+        """ Confirm it correctly runs on our example single record """
+        result = self.matcher\
+                     .code_record(title='Physicist',
+                                  sector='Professional scientific',
+                                  description='Calculations of the universe')
+
+        self.assertEqual(result, '211')
+
+    def test_code_data_frame(self):
         """Running the included examples from a file."""
-        myCoder = coder.Coder()
         df = pd.read_csv(os.path.join('tests', 'test_vacancies.csv'))
-        df = myCoder.codedataframe(df)
+        df = self.matcher.code_data_frame(df,
+                                          title_column="job_title",
+                                          sector_column="job_sector",
+                                          description_column="job_description")
         self.assertEqual(df['SOC_code'].to_list(), ['211', '242', '912'])
 
-    def test_002_command_line_use(self):
-        subprocess.run(['python', '-m',
-                        'occupationcoder.coder.coder',
+    # def test_parallel_code_data_frame(self):
+    #     """
+    #     Running the included examples from a file.
+    #     DISABLED because it can't be run through testr, parallel testing
+    #     interferes with parallelism in code
+    #     """
+    #     df = pd.read_csv(os.path.join('tests', 'test_vacancies.csv'))
+    #     df = self.matcher.parallel_code_data_frame(
+    #         df,
+    #         title_column="job_title",
+    #         sector_column="job_sector",
+    #         description_column="job_description"
+    #         )
+    #     self.assertEqual(df['SOC_code'].to_list(), ['211', '242', '912'])
+
+    def test_command_line(self):
+        """ Test code execution at command line """
+
+        # sys.executable returns current python executable, ensures code is run
+        # in same environment from which tests are called
+        subprocess.run([sys.executable, '-m',
+                        'occupationcoder.coder',
                         'tests/test_vacancies.csv'])
         df = pd.read_csv(os.path.join('occupationcoder',
                                       'outputs',
                                       'processed_jobs.csv'))
         self.assertEqual(df['SOC_code'].to_list(), [211, 242, 912])
+
+    def manual_load_test(self):
```

Comment:
Should we add test prefix to run as part of testing? As in `test_manual_load_test`